core/src/com/goonsquad/galactictd/systems/utils/SortedEntityComponentArray.java

```diff
@@ -42,34 +46,31 @@ public void insert(int entityId) {
 
     private void insert(int index, int entityId) {
         if (sortedEntityIds.length == size) expandSorted();
-        //Shift everything in the array to the right if it is after the new index to insert.
-        for (int i = size; i > index; i--) {
-            sortedEntityIds[i] = sortedEntityIds[i - 1];
-        }
+        //Shift everything in the array to the right if it is after the new index to insertValue.
+        System.arraycopy(sortedEntityIds, index, sortedEntityIds, index + 1, size - index);
         sortedEntityIds[index] = entityId;
         size++;
     }
 
-    public void remove(int entityId) {
+    public void removeValue(int entityIdToRemove) {
```

Comment:
Should we consider shrinking the array to half size if it goes below 1/4 of the size (or something like that)?